components/application/route.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Route is developed to simplify constructing routes within the application.
+ * It uses class instead a function to be extendible.
+ *
+ * Route should be considered being a string.
+ */
+class Route {
+  constructor(context) {
+    const params = typeof context == 'string' ? Route.parse(context) : context
+
+    const { dataProvider, activity } = params
+    this.dataProvider = dataProvider
+    this.activity = activity
+  }
+
+  toString() {
+    return [`/data-providers/${this.dataProvider}`, this.activity]
+      .filter(notEmpty => notEmpty)
+      .join('/')
+  }
+}
+
+Route.parse = routeString => {
```

Comment:
I wonder why this cannot be static method inside Route?

Reply:
That actually is. static keywords does the same I did here.

Here I use Route inside so I decided to write separate function to be clear that thinks I use here are declared above.